framework/src/modules/chain/dpos/delegates_info.js

```diff
@@ -162,24 +162,40 @@ class DelegatesInfo {
 		);
```

Comment:
This comment is related to the naming and responsibility of the function itself.

What does `_updateVotedDelegatesVoteWeight` do?
- Find delegates with earnings for this round (a separate function which is good)
- Find delegates that voted for other delegates
- Calculate the fees + reward
- Update the weight for votes

First of all, in my opinion, that's too much responsibility for a private function (bit out of scope, but as we are replacing it, why not think about this)

Next, the naming of the function does not include everything that happens inside. Actually, the last part that happens in the `.map` function is related to the function name `_updateVotedDelegatesVoteWeight` -> Therefore I suggest to split this.

Reply:
This is out of scope of this PR but a valid concern.

From your comment:
> Find delegates that voted for other delegates

This is just a filter to avoid a `if` inside the `map`.
But I agree that it can be a separated private function. Will update.


> Calculate the fees + reward

It doesn't calculate them.
The calculation happens inside `_getDelegatesWithTheirEarnings`


> that's too much responsibility for a private function

Apart from moving the `filter` to a separated function, what else could be extracted.

Reply:
> Calculate the fees + reward
Could use a wrapper function or pure function to do the summation. I think the final number should be coming in here and we shoudn't be doing any "calculations" inside here anymore (even a simple summation).

> What else could be extracted.
Possibly the content in `.map` function could be also private function that focuses only on changing the vote weight for delegates.

> But I agree that it can be a separated private function. Will update.
Danke!